### PR TITLE
Support passing URL as an argument to flag --file

### PR DESCRIPTION
**Pull request by programmer04**

### CODE REVIEW

File: pkg/discoverfiles/discoverfiles.go
Comment (nieomylnieja) [inline pkg/discoverfiles/discoverfiles.go:35]:
```suggestion
// is true. For "-" path that indicates standard input or starts with http:// or https://,
// it returns the path directly in the same way as other paths.
```

File: internal/pkg/yamlutils/yamlutils_test.go
Comment (nieomylnieja) [inline internal/pkg/yamlutils/yamlutils_test.go:16]: why?

**Review by nieomylnieja — CHANGES_REQUESTED** [review]

File: internal/pkg/yamlutils/yamlutils.go
Comment (programmer04) [reply internal/pkg/yamlutils/yamlutils.go:44]: We do not have proper context propagation, yet thus I made this shortcut by disabling it, but one day it should be enabled :)

File: internal/pkg/yamlutils/yamlutils_test.go
Comment (programmer04) [reply internal/pkg/yamlutils/yamlutils_test.go:16]: I'm in favor of the style described e.g. in [this great post](https://medium.com/@matryer/5-simple-tips-and-tricks-for-writing-unit-tests-in-golang-619653f90742#:~:text=Put%20your%20tests%20in%20a,real%20user%20of%20the%20package.)

File: internal/pkg/yamlutils/yamlutils.go
Comment (nieomylnieja) [reply internal/pkg/yamlutils/yamlutils.go:44]: I'd still prefer to comment out (with an explanation why) this linter in the linter config rather than do that

File: internal/pkg/yamlutils/yamlutils_test.go
Comment (nieomylnieja) [reply internal/pkg/yamlutils/yamlutils_test.go:16]: I get the premise now, It's actually an interesting point, I think I agree too :)
there's one thing from that post which boggles me though: "Run all tests on save" :D either this guy is working on very small projects or he's working on a supercomputer

Comment (programmer04) [reply internal/pkg/yamlutils/yamlutils_test.go:16]: Hehe, I think `oslo` fulfills the definition of a small project ;)

File: internal/pkg/yamlutils/yamlutils.go
Comment (nieomylnieja) [reply internal/pkg/yamlutils/yamlutils.go:44]: I get it, but on the other hand I don't think having that nolint there will ever prompt anyone to do sth about it either, I myself after 1.5 years of working with golangci-lint am completely numb to nolints, and as you said, we don't bother with context propagation yet, once we do, having a linter or not won't make a difference. I also don't quiet understand the point of having this linter in first place, from what I've seen it's very basic, it does not detect if the context you're passing actually does anything or if it's not empty (is there a deadline, cancellation on it). My experience with linters is that there are some which are very useful, and there's about an equal amount which are just a pain... I can't count how many hours I've wasted to nolint and satisfy the machine while waiting for CI/CD pipeline to re-validate the checks. I guess my point is, if you end up no-linting stuff it's probably a red light that either you shouldn't be adding these nolints and just fix the issue or the linter should not impose these rules on you, and rather you and the reviewer should decide case by case. But it's a longer rant and general beef I have with lint-all approach, I don't want to impose it here and now :) let's leave it with the `nolint`

**Review by nieomylnieja — APPROVED** [review]
looks good! kudos for adding the tests to cover some of these paths :)

**Review by nieomylnieja — APPROVED** [review]

**Review by fpiwowarczyk — APPROVED** [review]
Lgtm

**Pull request merged**
